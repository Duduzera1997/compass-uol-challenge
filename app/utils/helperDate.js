import { format, formatDistanceToNow as formatDistance } from 'date-fns';
import ptBR from 'date-fns/locale/pt-BR';

export function formatDateLiteral(date) {
  return format(new Date(date), "dd 'de' MMMM 'de' yyyy", {
    locale: ptBR,
  });
}

export function formatDistanceToNow(date) {
  return `${formatDistance(date, { locale: ptBR })} atr√°s`;
}

export function formatDateWithDistance(date) {
  return `${formatDateLiteral(date)} | ${formatDistanceToNow(date)}`;
}
