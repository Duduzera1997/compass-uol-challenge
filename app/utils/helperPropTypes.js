import PropTypes from 'prop-types';

export const ArrayOfArticlePropTypes = PropTypes.shape({
  last_updated: PropTypes.string,
  num_results: PropTypes.number,
  results: PropTypes.arrayOf(PropTypes.shape(ArticlePropTypes)).isRequired,
});

export const ArticlePropTypes = PropTypes.shape({
  section: PropTypes.string,
  subsection: PropTypes.string,
  title: PropTypes.string,
  abstract: PropTypes.string,
  url: PropTypes.string,
  uri: PropTypes.string,
  byline: PropTypes.string,
  item_type: PropTypes.string,
  updated_date: PropTypes.string,
  created_date: PropTypes.string,
  published_date: PropTypes.string,
  material_type_facet: PropTypes.string,
  kicker: PropTypes.string,
  des_facet: PropTypes.arrayOf(PropTypes.string),
  org_facet: PropTypes.arrayOf(PropTypes.string),
  per_facet: PropTypes.arrayOf(PropTypes.string),
  geo_facet: PropTypes.arrayOf(PropTypes.string),
  multimedia: PropTypes.arrayOf(PropTypes.shape(MultimidiaPropTypes)),
  short_url: PropTypes.string,
});

export const MultimidiaPropTypes = PropTypes.shape({
  url: PropTypes.string,
  format: PropTypes.string,
  height: PropTypes.integer,
  width: PropTypes.integer,
  type: PropTypes.string,
  subtype: PropTypes.string,
  caption: PropTypes.string,
  copyright: PropTypes.string,
});
