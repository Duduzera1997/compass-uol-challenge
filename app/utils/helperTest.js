import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import theme from '~/styles/theme';

export const renderWithTheme = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export const renderWithStoreAndThemeProviders = (component, { techMock }) => {
  const mockStore = configureStore();
  const state = {
    techReducer: {
      ...techMock,
    },
  };
  const store = mockStore(state);
  return renderWithTheme(<Provider store={store}>{component}</Provider>);
};
