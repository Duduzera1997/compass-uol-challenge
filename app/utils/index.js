export * from './constants';
export * from './helperDate';
export * from './helperTest';
export * from './helperPropTypes';
