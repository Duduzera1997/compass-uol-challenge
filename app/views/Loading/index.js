import React from 'react';
import { Loading } from '~/components';

function LoadingScreen() {
  return <Loading />;
}
export default LoadingScreen;
