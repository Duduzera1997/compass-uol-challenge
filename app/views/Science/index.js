import React from 'react';
import { View, Text } from 'react-native';

function ScienceScreen() {
  return (
    <View>
      <Text>Science</Text>
    </View>
  );
}

export default ScienceScreen;
