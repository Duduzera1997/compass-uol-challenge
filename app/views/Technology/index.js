import React, { useCallback, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import _ from 'lodash';

import * as SC from './styles';
import { techActions } from '~/redux';
import {
  LinearBackground,
  Header,
  ListArticles,
  ViewArticleModal,
  SearchInput,
} from '~/components';

function TechnologyScreen() {
  const { data, loading, error } = useSelector((state) => state.techReducer);
  const [filterText, setFilterText] = useState('');

  const [modalInfo, setModalInfo] = useState({ visible: false, item: null });
  const dispatch = useDispatch();

  useEffect(() => {
    fetchArticlesTechnology();
  }, [fetchArticlesTechnology]);

  const fetchArticlesTechnology = useCallback(
    () => dispatch(techActions.fetchTechnologiesAction()),
    [dispatch],
  );

  function getFilteredList() {
    if (_.isEmpty(filterText)) return data;
    const filteredResults = data.results.filter(
      (item) =>
        item.title.toLowerCase().includes(filterText.toLowerCase()) ||
        item.abstract.toLowerCase().includes(filterText.toLowerCase()),
    );
    return {
      ...data,
      results: filteredResults,
      num_results: filteredResults.length,
    };
  }

  function onCardPressed(item) {
    setModalInfo({ ...modalInfo, visible: true, item });
  }

  function onRequestCloseModal() {
    setModalInfo({ ...modalInfo, visible: false, item: null });
  }

  const filteredList = getFilteredList();

  return (
    <LinearBackground>
      <SC.ContainerTech>
        {modalInfo.visible && (
          <ViewArticleModal
            visible={modalInfo.visible}
            item={modalInfo.item}
            onRequestClose={onRequestCloseModal}
          />
        )}
        <SC.ContainerByFlex flex={1}>
          <SC.ContainerByFlex flex={1}>
            <Header headerText="Technology" />
          </SC.ContainerByFlex>
          {!!data?.results?.length && (
            <SC.ContainerByFlex flex={1}>
              <SearchInput onChangeText={_.debounce(setFilterText, 500)} />
            </SC.ContainerByFlex>
          )}
          <SC.ContainerByFlex flex={6}>
            <ListArticles
              data={filteredList}
              loading={
                loading || (!loading && !!filterText && !!filteredList.length)
              }
              error={error}
              onRefresh={fetchArticlesTechnology}
              onCardPress={onCardPressed}
            />
          </SC.ContainerByFlex>
        </SC.ContainerByFlex>
      </SC.ContainerTech>
    </LinearBackground>
  );
}

export default TechnologyScreen;
