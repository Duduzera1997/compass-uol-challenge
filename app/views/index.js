import LoadingScreen from './Loading';
import TechnologyScreen from './Technology';
import ScienceScreen from './Science';

export { LoadingScreen, TechnologyScreen, ScienceScreen };
