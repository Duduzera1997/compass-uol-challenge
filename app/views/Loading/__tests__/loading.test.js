import React from 'react';
import { renderWithTheme } from '~/utils';
import { LoadingScreen } from '~/views';

describe('Loading Screen', () => {
  it('should be render screen', () => {
    const { getByTestId, toJSON } = renderWithTheme(<LoadingScreen />);
    const logo = getByTestId('component-loading-logo');
    expect(logo).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
