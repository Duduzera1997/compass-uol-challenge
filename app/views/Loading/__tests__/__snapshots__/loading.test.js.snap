// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Screen should be render screen 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282066552,
      4294924895,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 1.05,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/img/logo-white.png",
        }
      }
      style={
        Array [
          Object {
            "height": 55,
            "width": 244,
          },
        ]
      }
      testID="component-loading-logo"
    />
    <ActivityIndicator
      animating={true}
      color="#FFFFFF"
      hidesWhenStopped={true}
      size="small"
      style={
        Array [
          Object {},
        ]
      }
      testID="component-loading-item"
    />
  </View>
</BVLinearGradient>
`;
